From b2358f24a93d5a07b5bf7c35f0d76676dc85869c Mon Sep 17 00:00:00 2001
From: Mario Limonciello <mario.limonciello@amd.com>
Date: Thu, 17 Oct 2024 11:44:20 -0500
Subject: [PATCH] acppdmmach: add support for ACP 7.0

Starting with ACP 7.0 there is a generic 'acppdmmach' driver that
will be utilized.

Closes: https://github.com/alsa-project/alsa-ucm-conf/pull/452
Signed-off-by: Mario Limonciello <mario.limonciello@amd.com>
Signed-off-by: Jaroslav Kysela <perex@perex.cz>
Origin: upstream, https://github.com/alsa-project/alsa-ucm-conf/commit/b2358f24a93d5a07b5bf7c35f0d76676dc85869c
Bug-Ubuntu: https://bugs.launchpad.net/bugs/2085067
Signed-off-by: Aaron Ma <aaron.ma@canonical.com>
---
 ucm2/HDA/HDA.conf                          | 2 +-
 ucm2/conf.d/acp-pdm-mach/acp-pdm-mach.conf | 1 +
 2 files changed, 2 insertions(+), 1 deletion(-)
 create mode 120000 ucm2/conf.d/acp-pdm-mach/acp-pdm-mach.conf

--- a/ucm2/HDA/HDA.conf
+++ b/ucm2/HDA/HDA.conf
@@ -3,7 +3,7 @@
 Define.Use ""	# a non-empty string to use UCM configuration for HDA devices
 Define.Done ""	# a non-empty string to skip the end error
 
-Define.AcpCardId "$${find-card:field=name,return=id,regex='^acp(|6[23x])$'}"
+Define.AcpCardId "$${find-card:field=name,return=id,regex='^acp(|6[23x]|-pdm-mach)$'}"
 Define.DeviceMic "Mic"
 
 If.dualcodec {
--- /dev/null
+++ b/ucm2/conf.d/acp-pdm-mach/acp-pdm-mach.conf
@@ -0,0 +1,3 @@
+Syntax 3
+
+Include.linked.File "/common/linked.conf"
